include/async/variant_sender.hpp

```diff
@@ -151,12 +151,29 @@ template <typename... Args> constexpr auto make_variant(Args &&...args) {
 }
 
 namespace _variant {
+template <std::size_t N, typename R, typename F, typename V>
+auto visit_workaround(F &&f, V &&v) -> R {
```

Comment:
suggestion: add a comment explaining why we need the workaround and a link to any clang bug filing

Reply:
:+1: I am not exactly sure what the bug is yet. There's at least one case of just not compiling that's hard to minimize. And it seems like there's some kind of issue in variant/visit that triggers TSan. Which is weird. I'll keep an eye on the clang bugs.